feg/gateway/services/session_proxy/credit_control/gx/config_test.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gx
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"magma/gateway/mconfig"
+)
+
+func TestGxClientConfig(t *testing.T) {
+	// Create tmp mconfig test file & load configs from it
+	fegConfigFmt := `{
+		"configsByKey": {
+			"session_proxy": {
+				"@type": "type.googleapis.com/magma.mconfig.SessionProxyConfig",
+				"gx": {
+					"servers": [{
+						"protocol": "sctp",
+						"address": "1.1.1.1:9999",
+						"retransmits": 5,
+						"watchdog_interval": 1,
+						"retry_count": 5,
+						"product_name": "magma_test",
+						"realm": "local.openair4G.eur",
+						"host": "local.magma-oai.openair4G.eur",
+						"dest_host": "magma-oai.openair4G.eur",
+						"dest_realm": "openair4G.eur",
+						"request_timeout": 10
+					}]
+				}
+			}
+		}
+	}`
+
+	err := mconfig.CreateLoadTempConfig(fegConfigFmt)
+	assert.NoError(t, err)
+	cliConfig := GetGxClientConfiguration()[0]
+
+	assert.Equal(t, "local.magma-oai.openair4G.eur", cliConfig.Host)
+	assert.Equal(t, "local.openair4G.eur", cliConfig.Realm)
+	assert.Equal(t, "magma_test", cliConfig.ProductName)
+	assert.Equal(t, uint(1), cliConfig.WatchdogInterval)
+	assert.Equal(t, uint(5), cliConfig.RetryCount)
+	assert.Equal(t, uint(5), cliConfig.Retransmits)
+	assert.Equal(t, uint(10), cliConfig.RequestTimeout)
```

Comment:
Can you make sure you cover all the fields 
```
			{
				Host:               diameter.GetValueOrEnv(diameter.HostFlag, GxDiamHostEnv, diameter.DiamHost),
				Realm:              diameter.GetValueOrEnv(diameter.RealmFlag, GxDiamRealmEnv, diameter.DiamRealm),
				ProductName:        diameter.GetValueOrEnv(diameter.ProductFlag, GxDiamProductEnv, diameter.DiamProductName),
				AppID:              diam.GX_CHARGING_CONTROL_APP_ID,
				WatchdogInterval:   diameter.DefaultWatchdogIntervalSeconds,
				RequestTimeout:     diameter.DefaultRequestTimeoutSeconds,
				RetryCount:         uint(retries),
				Retransmits:        uint(retransmits),
				SupportedVendorIDs: diameter.GetValueOrEnv("", GxSupportedVendorIDsEnv, ""),
			},
		}
		```

Reply:
I've added also test for values from env variables. Before read config I set env value and unset it right after test finished. I also thought about saving original value of env var, and set it back after test. What do you think is the most preferable way?

Reply:
This looks fine. I was more concerned about covering all the fields

If you want to test the other path, you can use `os.Setenv`